ChromeExtension/js/background.js

```diff
@@ -109,19 +109,25 @@ function createGeniusLink(url) {
                 action: "loading"
             }, function (response) {});
         });
-
     }
 
-
     var client = new GeniusLinkServiceClient('https://api.geni.us/v3', localStorage['apiKey'], localStorage['apiSecret']);
     client.postToService('shorturls', {
             GroupId: localStorage['defaultGroupId'],
             Domain: localStorage['selectedDomainName'],
             Url: url
         },
         function (data) {
-
-            newLink = data.ShortUrl.Domain + "/" + data.ShortUrl.BaseCode;
+            var domain = data.ShortUrl.Domain.toString();

```

Comment:
Shove that into its own function ya?

Reply:
not sure which part you want me to shove into its own function since they're different? The http/https part is either replaced or added (if doesn't already exist) for this one, whereas for the other one we're always prepending with http/https. So there's some extra if blocks for this one